#! /bin/sh -e

# DP: PR target/25343, fix gcc.dg/pch/pch for m68k.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

2008-04-27  Roman Zippel <zippel@linux-m68k.org>

        * config/host-linux.c (TRY_EMPTY_VM_SPACE): Define for __mc68000__.


Index: gcc/config/host-linux.c
===================================================================
--- gcc/config/host-linux.c
+++ gcc/config/host-linux.c
@@ -85,6 +85,8 @@
 # define TRY_EMPTY_VM_SPACE	0x8000000000
 #elif defined(__sparc__)
 # define TRY_EMPTY_VM_SPACE	0x60000000
+#elif defined(__mc68000__)
+# define TRY_EMPTY_VM_SPACE	0x40000000
 #else
 # define TRY_EMPTY_VM_SPACE	0
 #endif

